### DISCUSSION TITLE
In OAuth2.0 Protocol Redirect-Uri Must Not Include Fragment

### DISCUSSION THREAD

**Discussion by alihankuzucuk** [scalar/scalar#4001] [Q&A]
I am trying to develop Identity Server application which is use OpenIddict and Identity Client application which is use Scalar.

In Scalar interface i can use "Authorize" buttons in individual endpoints because they not includes fragment ( # ) in redirect_uri
but in the first section of Scalar UI, I cannot use "Authorize" button cause of it automatically includes fragment ( # ) in redirect_uri.

So OpenIddict with rightly throws an error.

I am leaving decoded redirect uri here to inspect:

![screenshot](https://github.com/user-attachments/assets/67f655a4-dd78-4a68-b292-3283496188c4)

http://localhost:5000/connect/authorize?response_type=token&client_id=internal-client&redirect_uri=http://localhost:5002/scalar/v1#tag/culture/GET/api/v1/culture/supported-cultures&scope=identity.authenticated+alihankuzucuk.manage&state=xy05vk

You can also see protocol documentation below for uri must not include fragment:

1. https://www.rfc-editor.org/rfc/rfc6749#section-3.1.2
2. https://datatracker.ietf.org/doc/html/rfc6749#section-3.1.2
3. https://github.com/openiddict/openiddict-core/issues/1554

**Comment by hanspagel** [comment]
You’re so right! Created an issue: https://github.com/scalar/scalar/issues/4003

**Reply by alihankuzucuk** [reply]
Thanks. I am closing this issue right now
